index/secondary-index/src/test/scala/org/apache/carbondata/spark/testsuite/secondaryindex/TestSIWithAddSegment.scala

```diff
@@ -86,8 +86,8 @@ class TestSIWithAddSegment extends QueryTest with BeforeAndAfterAll {
     sql(s"alter table maintable1 add segment options('path'='${ newSegmentPath }', " +
         s"'format'='carbon')")
     sql("CREATE INDEX maintable1_si  on table maintable1 (c) as 'carbondata'")
-    assert(sql("show segments for table maintable1_si").collect().length ==
-           sql("show segments for table maintable1").collect().length)
+    assert(sql("show segments for table maintable1_si").collect().length == 2)
+    assert(sql("show segments for table maintable1").collect().length == 3)
```

Comment:
Disabled SI table after alter add load and added a check to verify in test cases.

Reply:
@akashrn5 why we need to disable SI ?

Reply:
since for external added segments we don't load SI, its scenario of segment mismatch of SI and main table, it may lead to wrong results or failure. We don't yet support of getting the results from SI for some segments and some segments from main table right.

Reply:
We use externalSegment Resolver for pruning external segments on main table and for remaining segments, we create a implicit filter to prune using SI. So, there will not be any wrong results for the query